omnipaxos/src/omni_paxos.rs

```diff
@@ -67,31 +64,23 @@ impl OmniPaxosConfig {
             !self.peers.contains(&self.pid),
             "Peers should not include self pid"
         );
-
-        let num_nodes = self.peers.len() + 1;
-        match (self.leader_quorum_size, self.append_quorum_size) {
-            (Some(prepare_quorum_size), Some(accept_quorum_size)) => {
-                let quorums_overlap = prepare_quorum_size + accept_quorum_size > num_nodes;
-                assert!(
-                    quorums_overlap,
-                    "The sum of quorum sizes must exceed the # of nodes"
-                );
-                assert!(
-                    prepare_quorum_size > 0 && prepare_quorum_size <= num_nodes,
-                    "Prepare quorum must be in range 0 to # of nodes in the cluster"
-                );
-                // TODO: currently accept_quorum_size = 1 doesn't work, add to check?
-                assert!(
-                    accept_quorum_size > 0 && accept_quorum_size <= num_nodes,
-                    "Accept quorum must be in range 0 to # of nodes in the cluster"
-                );
-                assert!(prepare_quorum_size >= accept_quorum_size, "Leader election cannot guarantee progress if prepare quorum is smaller than accept quorum");
-            }
-            (None, Some(_)) => panic!("Prepare quorum must also be defined"),
-            (Some(_), None) => panic!("Accept quorum must also be defined"),
-            (None, None) => (),
+        if let Some((prepare_quorum_size, accept_quorum_size)) = self.flexible_quorum {
+            let num_nodes = self.peers.len() + 1;
+            let quorums_overlap = prepare_quorum_size + accept_quorum_size > num_nodes;
+            assert!(
+                quorums_overlap,
+                "The quorums must overlap i.e., the sum of their sizes must exceed the # of nodes"
+            );
+            assert!(
+                prepare_quorum_size >= 1 && prepare_quorum_size <= num_nodes,
+                "Read quorum must be in range 1 to # of nodes in the cluster"
+            );
+            assert!(
+                accept_quorum_size >= 2 && accept_quorum_size <= num_nodes,
+                "Write quorum must be in range 2 to # of nodes in the cluster"
+            );
+            assert!(prepare_quorum_size >= accept_quorum_size, "Leader election cannot guarantee progress if read quorum is smaller than write quorum");
```

Comment:
I think we should move these asserts to the `Quorum::with()` function. This way the checks will be performed regardless of where we build them i.e. OmniPaxosConfig, SequencePaxosConfig, BLEConfig etc.